packages/signature-v4/src/SignatureV4.spec.ts

```diff
@@ -452,6 +452,27 @@ describe("SignatureV4", () => {
       );
     });
 
+    it("should allow specifying custom signable headers to override custom and always unsignable ones", async () => {
+      const { headers } = await signer.sign(
+        {
+          ...minimalRequest,
+          headers: {
+            host: "foo.us-bar-1.amazonaws.com",
+            foo: "bar",
+            "user-agent": "baz"
+          }
+        },
+        {
+          signingDate: new Date("2000-01-01T00:00:00.000Z"),
+          unsignableHeaders: new Set(["foo"]),
+          signableHeaders: new Set(["foo", "user-agent"])
+        }
+      );
+      expect(headers[AUTH_HEADER]).toMatch(
+        /^AWS4-HMAC-SHA256 Credential=foo\/20000101\/us-bar-1\/foo\/aws4_request, SignedHeaders=host;foo;user-agent;x-amz-content-sha256;x-amz-date, Signature=/
```

Comment:
Edit to fix [CI failure](https://github.com/aws/aws-sdk-js-v3/pull/420#issuecomment-546516264):
```suggestion
        /^AWS4-HMAC-SHA256 Credential=foo\/20000101\/us-bar-1\/foo\/aws4_request, SignedHeaders=foo;host;user-agent;x-amz-content-sha256;x-amz-date, Signature=/
```

Reply:
I'm not able to commit this suggestion to repository.
* It could be because PR was created from master branch?
* The option ["allow edits from maintainers"](https://github.blog/2016-09-07-improving-collaboration-with-forks/) should have been set by default

![Screen Shot 2019-10-28 at 12 06 47 PM](https://user-images.githubusercontent.com/16024985/67709732-c4d38180-f97b-11e9-8059-46a94c85c78e.png)

![Screen Shot 2019-10-28 at 12 06 50 PM](https://user-images.githubusercontent.com/16024985/67709706-b84f2900-f97b-11e9-8ea9-73d65660c98b.png)